### PR TITLE
Improve error visibility

### PR DESCRIPTION
**Pull request by ahuang11**
## Summary

Based on @anna-geller feedback on https://github.com/PrefectHQ/prefect-dbt/pull/22 of running `dbt debug`.

To improve visibility, show stderr in exception rather than in a logger.exception statement, and will use last line of output as stderr if the stderr is missing.

Here's `ls something`
```
Traceback (most recent call last):
  File "/Users/andrew/mambaforge/envs/snowflake/lib/python3.9/site-packages/prefect/engine.py", line 888, in orchestrate_task_run
    result = await task.fn(*args, **kwargs)
  File "/Users/andrew/Applications/python/prefect-shell/prefect_shell/commands.py", line 56, in shell_run_command
    result = await utils_run_command(
  File "/Users/andrew/Applications/python/prefect-shell/prefect_shell/utils.py", line 85, in shell_run_command
    raise RuntimeError(msg)
RuntimeError: Command failed with exit code 1:
ls: something: No such file or directory

```

### Example
Code:
```
from prefect import task, flow
from prefect_shell import shell_run_command

@flow
def test_flow():
    shell_run_command("dbt debug")

if __name__ == "__main__":
    test_flow()
```

Previously:
```
09:44:45.049 | INFO    | Task run 'shell_run_command-7398b6ba-0' - dbt version: 1.1.1
python version: 3.9.12
python path: /Users/andrew/mambaforge/envs/snowflake/bin/python3.9
os info: macOS-11.6.5-arm64-arm-64bit
Using profiles.yml file at /Users/andrew/.dbt/profiles.yml
Using dbt_project.yml file at /Users/andrew/Applications/python/test/flow-of-flows/logs/dbt_project.yml

Configuration:
  profiles.yml file [OK found and valid]
  dbt_project.yml file [ERROR not found]

Required dependencies:
 - git [OK found]

Connection:

1 check failed:
Could not load dbt_project.yml


09:44:45.192 | ERROR   | Task run 'shell_run_command-7398b6ba-0' -
09:44:45.193 | ERROR   | Task run 'shell_run_command-7398b6ba-0' - Encountered exception during execution:
Traceback (most recent call last):
  File "/Users/andrew/mambaforge/envs/snowflake/lib/python3.9/site-packages/prefect/engine.py", line 888, in orchestrate_task_run
    result = await task.fn(*args, **kwargs)
  File "/Users/andrew/mambaforge/envs/snowflake/lib/python3.9/site-packages/prefect_shell/commands.py", line 56, in shell_run_command
    result = await utils_run_command(
  File "/Users/andrew/mambaforge/envs/snowflake/lib/python3.9/site-packages/prefect_shell/utils.py", line 81, in shell_run_command
    raise RuntimeError(msg)
RuntimeError: Command failed with exit code 1
09:44:45.207 | ERROR   | Task run 'shell_run_command-7398b6ba-0' - Finished in state Failed('Task run encountered an exception.')
```

Now:
```
python test.py
09:42:39.814 | INFO    | Task run 'shell_run_command-7398b6ba-0' - dbt version: 1.1.1
python version: 3.9.12
python path: /Users/andrew/mambaforge/envs/snowflake/bin/python3.9
os info: macOS-11.6.5-arm64-arm-64bit
Using profiles.yml file at /Users/andrew/.dbt/profiles.yml
Using dbt_project.yml file at /Users/andrew/Applications/python/test/flow-of-flows/logs/dbt_project.yml

Configuration:
  profiles.yml file [OK found and valid]
  dbt_project.yml file [ERROR not found]

Required dependencies:
 - git [OK found]

Connection:

1 check failed:
Could not load dbt_project.yml


09:42:39.964 | ERROR   | Task run 'shell_run_command-7398b6ba-0' - Encountered exception during execution:
Traceback (most recent call last):
  File "/Users/andrew/mambaforge/envs/snowflake/lib/python3.9/site-packages/prefect/engine.py", line 888, in orchestrate_task_run
    result = await task.fn(*args, **kwargs)
  File "/Users/andrew/mambaforge/envs/snowflake/lib/python3.9/site-packages/prefect_shell/commands.py", line 56, in shell_run_command
    result = await utils_run_command(
  File "/Users/andrew/mambaforge/envs/snowflake/lib/python3.9/site-packages/prefect_shell/utils.py", line 85, in shell_run_command
    raise RuntimeError(msg)
RuntimeError: Command failed with exit code 1:
Could not load dbt_project.yml
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 ### Changed
+- To improve visibility, show stderr in exception rather than in a logger.exception statement, and will use last line of output as stderr if the stderr is missing. - [#17](//github.com/PrefectHQ/prefect_shell/pull/17)
```
Comment (desertaxle) [inline CHANGELOG.md:13]: Can you add the description that you had to the PR description?
```suggestion
- Improve error visibility on failure - [#17](//github.com/PrefectHQ/prefect_shell/pull/17)
```

**Review by anna-geller — COMMENTED** [review]
I'm curious: is there a way to have both? Not sure if the error message would make it to the Prefect backend without the logger.error?

seeing exceptions in the UI is important for users

**Review by anna-geller — APPROVED** [review]
Nice work!

File: prefect_shell/utils.py [hunk 2]
```diff
@@ -76,8 +76,12 @@ async def shell_run_command(
...
                     [text async for text in TextReceiveStream(process.stderr)]
                 )
-                logger.error(stderr)
-                msg = f"Command failed with exit code {process.returncode}"
+                if not stderr and lines:
+                    stderr = f"{lines[-1]}\n"
+                msg = (
+                    f"Command failed with exit code {process.returncode}:\n" f"{stderr}"
+                )
+                logger.error(msg)
```
Comment (desertaxle) [inline prefect_shell/utils.py:84]: I tested and can see that the message in the `RuntimeError` does show up in the UI:
Screen Shot 2022-07-11 at 9 57 46 AM

The `logger.error` call results in the same message being logged twice, so I'd recommend removing this `logger.error` call.

**Review by desertaxle — CHANGES_REQUESTED** [review]

**Pull request merged**